articles/coding-on-purpose.md

```diff
@@ -0,0 +1,99 @@
+
+You might hear a lot of different goals from different people in the project. Especially if there's a lack of official goals, people might fill them up with their own hidden or not so hidden agenda. Try to figure out who spends money for the project and try to understand what outcomes they hope to achieve through this investment.
+
+### Lead Through Purpose
+
+Managers should put a strong focus on communicating the *why* of the things they ask you or a team to do and also judge your success by whether the *why* is fulfilled and not whether you built exactly *what* they asked for and did it *how* they expected you to do it. Simon Sinek would agree that communicating the *why* effectively increases the engagement of the team.
+
+### Coding Tips
+
+* If you are unsure whether what you are currently working on is actually needed for something meaningful, ask the relevant person (e.g. product owner) for the purpose of your work. It might turn out that it's not so important after all and that the stress you are experiencing is actually artificial and does not benefit anybody.
+
+* If you get told *what* to do it's usually good to ask back *what for*, as this gives you more flexibility in finding an economic solution for the desired outcome.
+
+* When coding, be aware that code does not document the *why* well. So be nice to your fellow coders and leave meaningful comments explaining the *why* or pointing to the relevant documentation.
+
+* When asking somebody for help, tell them what problem you are trying to solve, not just what problem you have with your current solution approach (see [The XY Problem](http://xyproblem.info/)).
+
+* And last but not least: no matter how minimalistic you want to keep your user stories, make sure they contain at least a goal and the desired benefit.
+
+
+## Thoughts On Customer And Supplier
+
+If you're building software, you are usually building it for a customer. This could be an internal department or another company. In an ideal world I would model the responsibilities regarding *what for* (purpose, goal), *what* (product, features) and *how* (development methodology, technology) like this:
+
+* The customer is responsible for knowing *what for* they want to do the project. They need an idea how to measure whether the desired outcome is achieved. And they need to be able to communicate the *purpose* to the supplier.
+
+* The supplier comes with a toolbox of skills. The know *how* to build software. They need to communicate to the customer what is hard and what is easy to build with this toolbox.
+
+* Now the customer and the supplier meet in the middle, one coming from the *what for* side and one from the *how* side. Together they need to agree on *what* they want to build. They optimize the *what* so that it achieves the customer goals (*what for*) while effectively utlizing the suppliers skills *how* to build it.
```

Comment:
that's it? I think the article needs a nice end - this last paragraph makes not a good end. I would as well say it is one of the worst paragraphs and I would leave it away (nothing new, nothing interessting in it, nothing that should not be obvious to everybody working in our business)

Reply:
In general I think if you shorten the article - it will get a good one! Just get to the point much faster, cause the interesting parts come only later.

I would focus on following parts and structure:
1. Purpose is important (very short intro: why should I read this? why is it important? What will I learn?)
2. Bad Examples not related to the Purpose
3. Good Examples more related to the Purpose
4. Practical Hints
5. Final Words: maybe ask the readers to tweet us about their thoughts about it, and what is their "why"?